src/hooks/useTxHistory.ts

```diff
@@ -11,16 +19,26 @@ const useTxHistory = (
   error?: string
   loading: boolean
 } => {
+  const router = useRouter()
+
   const { safe, safeAddress, safeLoaded } = useSafeInfo()
   const { chainId } = safe
 
   // If pageUrl is passed, load a new history page from the API
-  const [page, error, loading] = useAsync<TransactionListPage>(
+  const [page, error, loading] = useAsync<
+    | TransactionListPage
+    | SafeIncomingTransfersResponse
+    | SafeMultisigTransactionsResponse
+    | SafeModuleTransactionsResponse
+  >(
     () => {
       if (!pageUrl || !safeLoaded) return
```

Comment:
This check should include the filter now, otherwise it will work only for next pages.